### PR TITLE
Miscellaneous changes

### PR DESCRIPTION
**Pull request by wkaluza**
Collected cleanup commits from this week. Say goodbye to: some typos, one setter and a fair bit of unused code (including a few whole files).

### CODE REVIEW

**Review by n-hutton — APPROVED** [review]
Looks great

File: libs/network/include/network/generics/locked.hpp [hunk 1]
```diff
@@ -1,66 +0,0 @@
...
-//   See the License for the specific language governing permissions and
-//   limitations under the License.
-//
-//------------------------------------------------------------------------------
-
-namespace fetch {
-namespace generics {
-
-template <class TYPE, class MUTEXTYPE>
-class Locked
```
Comment (ejfitzgerald) [inline libs/network/include/network/generics/locked.hpp:24]: 👏 👏 👏

File: libs/network/include/network/generics/promise_of.hpp [hunk 2]
```diff
@@ -82,12 +82,6 @@ class PromiseOf : public ResolvableTo<RESULT>
     return promise_->name();
   }
 
-  // TODO(EJF): This seems a little scary, is this a copy or move?
-  void Adopt(Promise &promise)
```
Comment (ejfitzgerald) [inline libs/network/include/network/generics/promise_of.hpp:86]: 👏 👏 👏

File: libs/network/include/network/generics/shared_with_lock.hpp [hunk 3]
```diff
@@ -1,125 +0,0 @@
...
-//------------------------------------------------------------------------------
-
-#include <mutex>
-#include <utility>
-
-namespace fetch {
-namespace generics {
-
-template <class T>
-class SharedWithLock
```
Comment (ejfitzgerald) [inline libs/network/include/network/generics/shared_with_lock.hpp:27]: 👏 👏 👏

File: libs/network/include/network/generics/threadsafe_set.hpp [hunk 4]
```diff
@@ -1,116 +0,0 @@
...
-#include <set>
-#include <string>
-
-#include "network/generics/locked.hpp"
-
-namespace fetch {
-namespace generics {
-
-template <class TYPE>
-class ThreadsafeSet
```
Comment (ejfitzgerald) [inline libs/network/include/network/generics/threadsafe_set.hpp:30]: 👏 👏 👏

File: libs/vectorise/include/vectorise/threading/singleton_pool.hpp [hunk 5]
```diff
@@ -1,44 +0,0 @@
...
-//   limitations under the License.
-//
-//------------------------------------------------------------------------------
-
-#include "vectorise/threading/pool.hpp"
-
-namespace fetch {
-namespace threading {
-
-class SingletonPool : public Pool
```
Comment (ejfitzgerald) [inline libs/vectorise/include/vectorise/threading/singleton_pool.hpp:25]: I would double check with @baykaner that he isn't planning on using this in the future

**Review by ejfitzgerald — APPROVED** [review]
Just double check with @baykaner about SingletonPool

File: libs/vectorise/include/vectorise/threading/singleton_pool.hpp (hunk 5 above)
Comment (wkaluza) [reply libs/vectorise/include/vectorise/threading/singleton_pool.hpp:25]: He's fine with letting it die.

**Pull request merged**